File: OpenCv-Cpp/ScalingOperations/ScalingOperations.cpp
#include <iostream>
#include <string>
#include <opencv2/core.hpp>
#include <opencv2/highgui.hpp>
#include <opencv2/imgproc.hpp>

int main()
{
	std::string path;
	std::cout << "Give Path for Image: ";
	std::cin >> path;
	// Load Image to a Matrix
	cv::Mat image;
	image = cv::imread(path);

	// make sure data is loaded
	if (!image.data)
	{
		std::cout << "\nImage File data not loaded.\nExiting...";
		return -1;
	}

	// create empty matrix for storing half size image
	cv::Mat imageHalf;
	//compute to get half size image
	cv::resize(image, imageHalf, cv::Point(0, 0), 0.5, 0.5); // fx =0.5, fy=0.5

	// point variables for hard coded size
	cv::Point size = cv::Point(250, 250);

	// create empty matrix for storing image with hard coded size
	cv::Mat imageHardCode;
	//Hard coded resize
	cv::resize(image, imageHardCode, size);

	// create empty matrix for storing image with interpolated resize
	cv::Mat imageNearest;
	// resize with interpolation(Nearest Neighbor)
	cv::resize(image, imageNearest, size, cv::INTER_NEAREST); // interpolation = cv::INTER_NEAREST

	// Display Matrix(image) in GUI.
	cv::imshow("Original Image", image);
	cv::imshow("Half-Size", imageHalf);
	cv::imshow("Hard-Code", imageHardCode);
	cv::imshow("Interpolation(Nearest Neighbor)", imageNearest);

	// wait with delay for any key from GUI.
	cv::waitKey(0);
	cv::destroyAllWindows();
	return 0;
}

File: OpenCv-Python/ScalingOperations/ScalingOperations.py
import cv2

path = input("Enter path for Image:")

# Read image matrix from Image File
image = cv2.imread(path)

# check if image is read
if image is None:
    print('Image File data not loaded.\nExiting...')
    exit()

# compute to get half size image
image_half = cv2.resize(image, (0, 0), fx=0.5, fy=0.5)

# Hard coded resize
image_hardCode = cv2.resize(image, (250, 250))

# resize with interpolation (Nearest Neighbor)
image_nearest = cv2.resize(image, (250, 250), interpolation=cv2.INTER_NEAREST)

# show the images in a GUI
cv2.imshow('original', image)
cv2.imshow('Half-Size', image_half)
cv2.imshow('Hard-Code', image_hardCode)
cv2.imshow('Interpolation(Nearest Neighbor)', image_nearest)

# after delay wait for any key from GUI
cv2.waitKey(0)

# destroy or free any remaining GUI functionality
cv2.destroyAllWindows()
